test/api/iterations_test.rb

```diff
@@ -132,4 +89,56 @@ def test_skip_problem_as_guest
     assert_equal 401, last_response.status
     assert last_response.body.include?(expected_message)
   end
+
+  private
+
+  def create_submission!(user: nil)
```

Comment:
For now I just moved creation of objects from main tests here. It's probably better than but still not ideal. What about using something like FactoryGirl for this?